src/python/pants/backend/python/tasks/unpack_wheels.py

```diff
@@ -73,22 +86,18 @@ class WheelUnpackingError(TaskError): pass
   def unpack_target(self, unpacked_whls, unpack_dir):
     interpreter = self._compatible_interpreter(unpacked_whls)
 
-    with temporary_dir() as resolve_dir,\
-         temporary_dir() as extract_dir:
+    with temporary_dir() as resolve_dir:
       try:
         matched_dist = self._get_matching_wheel(resolve_dir, interpreter,
                                                 unpacked_whls.all_imported_requirements,
                                                 unpacked_whls.module_name)
-        ZIP.extract(matched_dist.location, extract_dir)
+        wheel_chroot = matched_dist.location
         if unpacked_whls.within_data_subdir:
-          data_dir_prefix = '{name}-{version}.data/{subdir}'.format(
-            name=matched_dist.project_name,
-            version=matched_dist.version,
-            subdir=unpacked_whls.within_data_subdir,
-          )
-          dist_data_dir = os.path.join(extract_dir, data_dir_prefix)
+          # N.B.: Wheels with data dirs have the data installed under the top module.
```

Comment:
I would also then deprecate the string form of the `within_data_subdir` kwarg in `UnpackedWheels`, something like:
```diff
diff --git a/src/python/pants/backend/python/targets/unpacked_whls.py b/src/python/pants/backend/python/targets/unpacked_whls.py
index 960c25396..f960c6dcd 100644
--- a/src/python/pants/backend/python/targets/unpacked_whls.py
+++ b/src/python/pants/backend/python/targets/unpacked_whls.py
@@ -6,6 +6,7 @@ import logging
 from twitter.common.collections import maybe_list

 from pants.backend.python.targets.import_wheels_mixin import ImportWheelsMixin
+from pants.base.deprecated import deprecated_conditional
 from pants.base.payload import Payload
 from pants.base.payload_field import PrimitiveField
 from pants.build_graph.target import Target
@@ -48,15 +49,21 @@ class UnpackedWheels(ImportWheelsMixin, Target):
     :param compatibility: Python interpreter constraints used to create the pex for the requirement
                           target. If unset, the default interpreter constraints are used. This
                           argument is unnecessary unless the native code depends on libpython.
-    :param str within_data_subdir: If provided, descend into '<name>-<version>.data/<subdir>' when
-                                   matching `include_patterns`. For python wheels which declare any
-                                   non-code data, this is usually needed to extract that without
-                                   manually specifying the relative path, including the package
-                                   version. For example, when `data_files` is used in a setup.py,
-                                   `within_data_subdir='data'` will allow specifying
-                                   `include_patterns` matching exactly what is specified in the
-                                   setup.py.
+    :param bool within_data_subdir: If True, descend into '<name>-<version>.data/' when matching
+                                    `include_patterns`. For python wheels which declare any non-code
+                                    data, this is usually needed to extract that without manually
+                                    specifying the relative path, including the package version. For
+                                    example, when `data_files` is used in a setup.py,
+                                    `within_data_subdir=True` will allow specifying
+                                    `include_patterns` matching exactly what is specified in the
+                                    setup.py.
     """
+    deprecated_conditional(
+      lambda: type(within_data_subdir) not in (bool, type(None)),
+      removal_version='1.26.0.dev2',
+      entity_description='A non-boolean value for `within_data_subdir`',
+      hint_message='The location of the .data subdirectory will be inferred from the module name!',
+    )
     payload = payload or Payload()
     payload.add_fields({
       'library_specs': PrimitiveField(libraries or ()),
```

Reply:
I've applied this diff. I'm not sure the help verbage is correct - `<name>-<version>.data/` simply never exists in an installed wheel chroot. But I'll save tightening all this up for a follow-up.